src/DynamoUtilities/Md2Html.cs

```diff
@@ -93,53 +96,63 @@ protected static string GetToolPath(string relativePath)
-                var done = false;
-                var start = false;
-                while (!done)
+                if (process.HasExited)
+                {
+                    return string.Empty;
+                }
+
+                using (var writer = new StringWriter())
                 {
-                    try
+                    var done = false;
+                    var start = false;
+                    while (!done)
                     {
-                        var line = process.StandardOutput.ReadLine();
-                        MessageLogged?.Invoke(line);
-                        if (line == null || line == startofDataToken)
+                        try
                         {
-                            start = true;
-                            continue;//don't record start token to stream.
-                        }
-                        if (line == null || line == endOfDataToken)
-                        {
-                            done = true;
-                        }
-                        else
-                        {   //if we have started recieving valid data, start recording
-                            if (!string.IsNullOrWhiteSpace(line) && start)
+                            var line = process.StandardOutput.ReadLine();
+                            MessageLogged?.Invoke(line);
+                            if (line == null || line == startofDataToken)
+                            {
+                                start = true;
+                                continue; //don't record start token to stream.
+                            }
+
+                            if (line == null || line == endOfDataToken)
+                            {
+                                done = true;
+                            }
+                            else
                             {
-                                writer.WriteLine(line);
+                                //if we have started recieving valid data, start recording
+                                if (!string.IsNullOrWhiteSpace(line) && start)
+                                {
+                                    writer.WriteLine(line);
+                                }
                             }
                         }
+                        catch (Exception)
+                        {
+                            KillProcess();
+                            return GetCantCommunicateErrorMessage();
+                        }
                     }
-                    catch (Exception)
-                    {
-                        KillProcess();
-                        return GetCantCommunicateErrorMessage();
-                    }
-                }
 
-                return writer.ToString();
-            }
+                    return writer.ToString();
+                }
+            });
+            var completedTask = await Task.WhenAny(readStdOutTask, Task.Delay(TimeSpan.FromMilliseconds(timeoutms)));
```

Comment:
so basically we are blocking a threadpool thread for 2 seconds in the worst case scenario ?

Reply:
well yes, we block a thread pool thread, but we are using `Task.Result` so if I'm understanding correctly we're blocking the main thread as well.

Reply:
For FF case, the getData call is actually on a thread pool thread, I think for Md2html it's on the main thread.